frontend/src/main/pages/RiderApplication/RiderApplicationIndexPageMember.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { useBackend } from 'main/utils/useBackend';
+
+import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+import {useCurrentUser} from 'main/utils/currentUser'
+import Button from 'react-bootstrap/Button';
+import { Link } from 'react-router-dom';
+import RiderApplicationTable from "main/components/RiderApplication/RiderApplicationTable";
+
+export default function RiderApplicationIndexPage() {
+
+    const currentUser = useCurrentUser();
+    // Stryker disable all
+    const currentUserCopy = { ...currentUser };
+    if (currentUserCopy.data?.root?.rolesList.includes("ROLE_ADMIN"))
+    {
+        currentUserCopy.data.root.rolesList = currentUserCopy.data.root.rolesList.filter(role => role !== "ROLE_ADMIN");
+    }
```

Comment:
This breaks react state immutability rules. You're making a shallow copy of currentUser (which is good, it shows you're conscious about avoiding side effects), but it's only a shallow copy. You must also copy any nested objects you want to modify or it **will** cause bugs in seemingly unrelated areas and it will be very difficult to debug.

Immer makes working with deep immutable data structures really easy. But since the class is finishing, to get this at least in a mergeable state this is the simplest fix.
```suggestion
    // TODO: use immer or related library to work with deep immutable data structures
    const currentUserCopy = currentUser.data?.root?.rolesList
        ? {
            ...currentUser,
            data: {
                ...currentUser.data,
                root: {
                    ...currentUser.data.root,
                    rolesList: currentUser.data.root.rolesList.filter(role => role !== "ROLE_ADMIN")
                }
            }
        }
        : { ...currentUser };
```

Reply:
Thank you for pointing it out! I've modified my code as you suggested and made a new push into this PR. Please give it a second review when you have time.